django/db/backends/sqlite3/base.py

```diff
@@ -408,6 +409,26 @@ def _sqlite_datetime_parse(dt, tzname=None, conn_tzname=None):
     if conn_tzname:
         dt = dt.replace(tzinfo=pytz.timezone(conn_tzname))
     if tzname is not None and tzname != conn_tzname:
+        _offset = None
+        for seperator in '+', '-':
+            if seperator in tzname:
+                tzname, _offset = tzname.split(seperator)
+                break
+
+        if _offset:
+            for _format in get_format('TIME_INPUT_FORMATS'):
+                try:
+                    _offset_date = datetime.datetime.strptime(_offset, _format)
+                    time_delta_params = {}
+                    for attr in {'second', 'minute', 'hour', 'millisecond'}:
+                        if hasattr(_offset_date, attr):
+                            attr_ = getattr(_offset_date, attr)
+                            time_delta_params["%ss" % attr] = attr_ if seperator == '+' else -attr_
+
+                    dt += datetime.timedelta(**time_delta_params)
+                    break
+                except ValueError:
+                    pass
```

Comment:
IMO SQLite solution is too complicated and contains a lot of unnecessary workarounds. Please find below my proposition:
```python
def _sqlite_datetime_parse(dt, tzname=None, conn_tzname=None):
    ...
    if tzname is not None and tzname != conn_tzname:
        for sign in '+', '-':
            if sign in tzname:
                tzname, offset = tzname.split(sign)
                if offset:
                    hours, minutes = offset.split(':')
                    offset_delta = datetime.timedelta(hours=int(hours), minutes=int(minutes))
                    dt += offset_delta if sign == '+' else -offset_delta
                break
        dt = timezone.localtime(dt, pytz.timezone(tzname))
    return dt
```

Probably it could be even simpler.

Reply:
I used find char index for '+', '-' to get rid of for loop at least.